recording-oracle/src/modules/campaigns/progress-checking/threshold.spec.ts

```diff
@@ -0,0 +1,233 @@
+import { faker } from '@faker-js/faker';
+import { createMock } from '@golevelup/ts-jest';
+
+import {
+  ExchangeApiClient,
+  ExchangeApiClientFactory,
+} from '@/modules/exchange';
+import { generateAccountBalance } from '@/modules/exchange/fixtures';
+
+import {
+  generateThresholdCheckerSetup,
+  generateParticipantAuthKeys,
+} from './fixtures';
+import { ThresholdProgressChecker } from './threshold';
+import { CampaignProgressCheckerSetup, ParticipantAuthKeys } from './types';
+
+const mockedExchangeApiClient = createMock<ExchangeApiClient>();
+const mockedExchangeApiClientFactory = createMock<ExchangeApiClientFactory>();
+
+class TestCampaignProgressChecker extends ThresholdProgressChecker {
+  override ethDepositAddresses = new Set<string>();
+}
+
+describe('ThresholdProgressChecker', () => {
+  beforeEach(() => {
+    mockedExchangeApiClientFactory.create.mockReturnValue(
+      mockedExchangeApiClient,
+    );
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should be defined', () => {
+    const resultsChecker = new TestCampaignProgressChecker(
+      mockedExchangeApiClientFactory as ExchangeApiClientFactory,
+      generateThresholdCheckerSetup(),
+    );
+    expect(resultsChecker).toBeDefined();
+  });
+
+  describe('checkForParticipant', () => {
+    let progressCheckerSetup: CampaignProgressCheckerSetup;
+    let participantAuthKeys: ParticipantAuthKeys;
+
+    let resultsChecker: TestCampaignProgressChecker;
+
+    beforeEach(() => {
+      progressCheckerSetup = generateThresholdCheckerSetup();
+      participantAuthKeys = generateParticipantAuthKeys();
+
+      resultsChecker = new TestCampaignProgressChecker(
+        mockedExchangeApiClientFactory as ExchangeApiClientFactory,
+        progressCheckerSetup,
+      );
+
+      mockedExchangeApiClient.fetchBalance.mockResolvedValue(
+        generateAccountBalance([
+          faker.finance.ethereumAddress(),
+          faker.finance.ethereumAddress(),
```

Comment:
nit
```suggestion
          faker.finance.currencyCode(),
          faker.finance.currencyCode(),
```
There is same type in holding campaign tests